src/Illuminate/View/Compilers/ComponentTagCompiler.php

```diff
@@ -312,6 +312,10 @@ public function componentClass(string $component)
             return $guess;
         }
 
+        if (Str::startsWith($component, 'mail::') && $viewFactory->exists($component)) {
+            return $component;
+        }
```

Comment:
I've given special treatment to `mail::` prefixed components, as they are given special treatment in the framework.